include/deal.II/fe/mapping_q.h

```diff
@@ -69,6 +69,53 @@ DEAL_II_NAMESPACE_OPEN
  * cells; again this class is then equivalent to using MappingQGeneric right
  * away.
  *
+ * <h4>Behavior along curved boundaries and with different manifolds</h4>
+ *
+ * As described above, one often only knows a manifold description of a
+ * surface but not the interior of the computational domain. In such a case, a
+ * StraightBoundary object will be assigned to the interior entities that
+ * describes a usual planar coordinate system where the additional points for
+ * the higher order mapping are placed exactly according to a bi-/trilinear
+ * mapping. When combined with a non-flat manifold on the boundary, for
+ * example a circle, the two manifold descriptions are in general
+ * incompatible. For example, a StraightBoundary defined solely through the
+ * cell's vertices would put an interior point located at some small distance
+ * epsilon away from the boundary along a flat line and thus in general
+ * outside the concave part of a circle. If the polynomial degree of
+ * MappingQGeneric is sufficiently high, the transformation from the reference
+ * cell to such a cell would in general contain inverted regions close to the
+ * boundary.
+ *
+ * In order to avoid this situation, this class applies a smoothing on cells
+ * adjacent to the boundary by using so-called Laplace smoothing by
+ * default. In the algorithm that computing additional points, the
+ * compute_mapping_support_points() method, all the entities of the cells are
+ * passed through hierarchically, starting from the lines to the quads and
+ * finally hexes. The elements higher up in the hierarchy that sit on the
+ * boundary will then get their points interpolated from all the surrounding
+ * points and not just the corner points. If only a line is assigned a curved
+ * boundary but the adjacent quad is on a flat manifold, the points inside the
+ * quad will be computed according to the deformed line and thus always result
+ * in a well-defined transformation. This smoothing can be disabled by setting
+ * the optional argument @p smooth_support_points to false, placing the
+ * additional points strictly according to the manifold. This is usually the
+ * most efficient choice in case different manifolds are present that are
+ * compatible with each other.
+ *
+ * While the smoothing approach works well for filling holes or avoiding
+ * inversions with low and medium convergence orders up to approximately three
+ * to four, there is nonetheless an inherent shortcoming because of a
+ * discontinuous mapping that switches from a curved manifold to a flat
+ * manifold within one layer of elements. This will cause the Jacobian
```

Comment:
I was a bit unclear here: The problem is not that the Jacobian matrix of the transformation is discontinuous but actually already the transformation itself: If I mix a curved description along a line with a flat description inside the element, the prescribed transformation makes a jump. In terms of the finite element theory (I read sections 2.2 and 3.3 in Strang and Fix), this is problematic because the mapping over element boundaries needs to be **globally** continuous for H^1-conforming problems, also over element boundaries. Even though we smooth the jump over a length `h`, the smoothing occurs in one layer of elements and the underlying transformation is thus not uniformly C^0. Thus, the theory cannot guarantee optimal convergence.